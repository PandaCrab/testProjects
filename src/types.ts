import { store } from './redux/store';

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

export type coordinates = {
    lat: number,
    lon: number
}

export type endpoint = {
    value: string
}

export type addressState = {
    addressInput: string,
    addresses: any,
    geolocation: {
        lat: number,
        lon: number
    },
    navigatorAddress: any
}

export type postData = {
    name: string,
    phone: number,
    email: string,
    street: string,
    optional: string,
    city: string,
    country: string,
    zip: number,
    cardHolder: string,
    cardNum: number,
    date: number,
    code: number
};

export type addressAutocomplete = {
    street?: string,
    city?: string,
    country?: string
}
