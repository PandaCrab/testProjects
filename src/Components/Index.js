import { ShippingInfo } from './Forms/ShippingInfo';
import { BillingInfo } from './Forms/BillingInfo';
import { Payment } from './Forms/Payment';
import { OrderPlate } from './OrderPlate/OrderPlate';


export { ShippingInfo, BillingInfo, Payment, OrderPlate };
