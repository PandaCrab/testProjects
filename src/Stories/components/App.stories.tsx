import React from 'react';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import { ApolloProvider } from '@apollo/client';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

import { store } from '../../redux/store';
import App from '../../App';
import { apolloClient } from '../..';

export default {
    title: 'App',
    component: App
} as ComponentMeta<typeof App>;

const Template: ComponentStory<typeof App> = () => (
    <ApolloProvider client={apolloClient}>
        <Provider store={store}>
            <BrowserRouter>
                <App />
            </BrowserRouter>
        </Provider>
    </ApolloProvider>
);

export const Header = Template.bind({});
