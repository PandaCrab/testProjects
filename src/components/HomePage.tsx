import React from 'react';
import { useNavigate } from 'react-router-dom';

import { 
    StartPage,
    NavigationButton
} from '../styles/HomePageStyles';

const HomePage = () => {
    const navigate = useNavigate();

    return (
        <>
            <StartPage>
                <h1>Hello World</h1>
                <NavigationButton onClick={() => navigate('/shopping')}>Start shopping</NavigationButton>
            </StartPage>
        </>
    );
};

export default HomePage;
