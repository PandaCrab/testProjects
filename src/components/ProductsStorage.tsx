import { useState } from 'react';
import { 
    SortingState,
    IntegratedSorting,
    RowDetailState
} from '@devexpress/dx-react-grid';
import {
  Grid,
  Table,
  TableHeaderRow,
  TableRowDetail,
} from '@devexpress/dx-react-grid-material-ui';
// import { useDispatch, useSelector } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import { useQuery } from '@apollo/client';

// import { AppDispatch, RootState } from '../types';

// import { getProductsStorage } from '../redux/ducks/stuff';
import { Heading, NavigationButton, StartPage } from '../styles/HomePageStyles';
import { TAKE_PRODUCTS_FROM_STORAGE } from '../api';

interface graphQlData {
    id: string,
    name: string,
    price: number,
    quantity: number
}

const RowDetail = ({ row }: any) => (
    <div>
        {row.quantity > 0 ? 
        `This item can be send today` :
        `Check later, out of stock`
        }
    </div>
)

const ProductsStorage = () => {
    const { error, data } = useQuery(TAKE_PRODUCTS_FROM_STORAGE);
    const navigate = useNavigate();
    // const dispatch = useDispatch<AppDispatch>();
    // const productsStorage = useSelector((state: RootState) => state.order.productsStorage);

    // useEffect(() => { 
    //     dispatch(getProductsStorage())
    // // eslint-disable-next-line react-hooks/exhaustive-deps
    // }, []);

    const [columns] = useState([
        { name: 'id', title: 'Id' },
        { name: 'name', title: 'Name' },
        { name: 'price', title: 'Price' },
        { name: 'quantity', title: 'Quantity' }
    ]);

    const [tableColumnExtensions] = useState([
        { columnName: 'id', width: '10%' },
        { columnName: 'name', width: '50%' },
        { columnName: 'price', width: '10%' },
        { columnName: 'quantity', width: '15%' },
      ]);

    return (
        <StartPage>
            <Heading>
                <h1>Hello in storage</h1>  <NavigationButton onClick={() => navigate('/')} >Home</NavigationButton>
            </Heading>
            <Grid
                rows={data.productsStorage.map(({ id, name, price, quantity }: graphQlData) => ({
                    id,
                    name,
                    price,
                    quantity
                }))}
                columns={columns}
            >
                <SortingState
                    defaultSorting={[{ columnName: 'name', direction: 'asc' }]}
                />
                <IntegratedSorting />
                <RowDetailState />

                <Table columnExtensions={tableColumnExtensions} />
                <TableHeaderRow showSortingControls />
                <TableRowDetail 
                    contentComponent={RowDetail}
                />
            </Grid>
        </StartPage>
    );
};

export default ProductsStorage;
