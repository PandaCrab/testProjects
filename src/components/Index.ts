import ShippingInfo from './forms/ShippingInfo';
import BillingInfo from './forms/BillingInfo';
import Payment from './forms/Payment';
import OrderPlate from './orderPlate/OrderPlate';
import SuccessBuying from './forms/SuccessBuying';
import Navigate from '../Navigate';
import Stuff from './orderPlate/Stuff';
import StuffItems from './orderPlate/StuffItems';
import Loader from './Loader';
import CardToPrint from './forms/CardToPrint';
import DropdownAddresses from './forms/dropdownAddresses';
import CountriesSelect from './forms/CountriesSelect';
import HomePage from './HomePage';
import ProductsStorage from './ProductsStorage';
import SaleChart from './SaleChart';

export { ShippingInfo,
    BillingInfo, 
    Payment, 
    OrderPlate, 
    SuccessBuying, 
    Navigate,
    Stuff,
    StuffItems,
    Loader,
    CardToPrint,
    DropdownAddresses,
    CountriesSelect,
    HomePage,
    ProductsStorage,
    SaleChart
};
