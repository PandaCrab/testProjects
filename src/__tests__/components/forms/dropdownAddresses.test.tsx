import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';

import { DropdownAddresses } from '../../../components/Index';
import { addressPayload } from '../../../types';

const initialState: { address: addressPayload } = {address: {
    addresses: [
        {
            properties: {
                id: 12234,
                name: '7a heros of Ukraine',
                locality: 'Lokvyts`a',
                country: 'Ukraine',
                label: '7a heros of Ukraine, Lokhvitsy`a, Ukraine'
            }
        },
        {
            properties: {
                id: 3552,
                name: '35 Ivana Sirka',
                locality: 'Sumy',
                country: 'Ukraine',
                label: '35 Ivana Sirka, Sumy, Ukraine'
            }
        }
    ]
}};
const mockStore = configureStore();
let store = mockStore(initialState);

const props = {
    autocomplete: jest.fn()
};

const component = mount(
    <Provider store={store}>
        <DropdownAddresses {...props} />
    </Provider>
);

describe('Dropdown addresses section render', () => {
    it('should render component correct', () => {
        expect(component).toMatchSnapshot();
    });

    it('should have items', () => {
        expect(component.find('li[key=12234]')).toBeDefined();
    });
});

describe('check activate autocomplete on click', () => {
    it('should fire autocomplete on click', () => {
        component.find('li').first().simulate('click');

        expect(props.autocomplete).toHaveBeenCalled();
    });
});
