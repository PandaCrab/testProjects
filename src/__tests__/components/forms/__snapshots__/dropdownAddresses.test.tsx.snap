// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown addresses section render should render component correct 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <DropdownAddresses
    autocomplete={[MockFunction]}
  >
    <styled.ul>
      <ul
        className="sc-gsnTZi iiYaAR"
      >
        <styled.li
          key="12234"
          onClick={[Function]}
        >
          <li
            className="sc-eCYdqJ hCLdSg"
            onClick={[Function]}
          >
            7a heros of Ukraine, Likhvitsy\`a, Ukraine
          </li>
        </styled.li>
        <styled.li
          key="3552"
          onClick={[Function]}
        >
          <li
            className="sc-eCYdqJ hCLdSg"
            onClick={[Function]}
          >
            35 Ivana Sirka, Sumy, Ukraine
          </li>
        </styled.li>
      </ul>
    </styled.ul>
  </DropdownAddresses>
</Provider>
`;
